virttest/qemu_devices/qdevices.py

```diff
@@ -1277,6 +1310,64 @@ def verify_unplug(self, out, monitor):
         return not self._is_attached_to_qemu(monitor)
 
 
+class QThrottleGroup(QObject):
+    """
+    throttle-group object.
+    """
+
+    def __init__(self, group_id, props):
+        self._raw_limits = props.copy()
+
+        params = QThrottleGroup._map_key(self.raw_limits)
+        params["id"] = group_id
+        kwargs = dict(backend="throttle-group", params=params)
+        super(QThrottleGroup, self).__init__(**kwargs)
+        self.throttle_group_bus = QThrottleGroupBus(group_id)
+        self.add_child_bus(self.throttle_group_bus)
+        self.set_aid(group_id)
+        self.hook_drive_bus = None
+
+    @staticmethod
+    def _map_key(params):
+        return {'x-%s' % k: v for k, v in params.items()}
```

Comment:
https://github.com/avocado-framework/avocado-vt/pull/2689#discussion_r473074025 @YongxueHong @qingwangrh the problem is about how we could make this compatible for future versions, the prefix may be dropped therefore we would have to change the approach again. To avoid that, there are two possible approaches:

1. do not do any conversion, just use the values specified by users, and users can take the control by using different configurations
2. append the prefix if necessary

both are fine to me, please let me know your opinion, thanks.

Reply:
Can't agree with that because, for example, if the prefix `x-` would be removed for some options of throttle-groups by future versions the code could not work properly any more and you have to update it to respect the new usage, that's why I prefer either

* not using conversion, then the code doesn't request a change (at least for VT part)

or

* a better approach which can handle the conversion correctly even for future versions